pkg/admin/prerun/upgrade_blocking.go

```diff
@@ -0,0 +1,59 @@
+package prerun
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+
+	embedded "github.com/openshift/microshift/assets"
+	"k8s.io/klog/v2"
+)
+
+func IsUpgradeBlocked(execVersion versionMetadata, dataVersion versionMetadata) (bool, error) {
+	buf, err := getBlockedUpgradesAsset()
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to load embedded blocked upgrades asset: %w", err)
+	}
+
+	m, err := unmarshalBlockedUpgrades(buf)
+	if err != nil {
+		return false, err
+	}
+
+	return isBlocked(m, execVersion.String(), dataVersion.String()), nil
+}
+
+func getBlockedUpgradesAsset() ([]byte, error) {
```

Comment:
Is using an asset easier to manage than just embedding the list in this go module? Are you thinking of automation managing that file (in which case, yes, I agree YAML is going to be easier)?

Reply:
At this point I don't have any idea how we'll use this feature and manage the list. Asset just felt natural since we already embed some files that way but it's not a strong feeling and we can change it